### DISCUSSION TITLE
Poll | Feedback Required | Should we make all PostgreSQL dependencies optional ?

### DISCUSSION THREAD

**Discussion by funilrys** [funilrys/PyFunceble#400] [Polls]
Since #398 has been submitted, it occured to me that not everyone needs those dependencies. Therefore, I propose to make them optional. 

For anyone that want to test how it will look like, please report to the [issue-398 branch](https://github.com/funilrys/PyFunceble/tree/issue-398). 

## What does this mean ?

This mean that once the [issue-398 branch](https://github.com/funilrys/PyFunceble/tree/issue-398) is merged, you will be required to explicitly install the PostgreSQL dependencies through the `psql` or `psql-binary` extras.

Here how it will look like:
```shell
# psql required.
pip3 --user --install PyFunceble-dev[psql-binary]
```

If you want to test - earlier, you can run:

```shell
pip3 --user --install PyFunceble-dev[psql-binary] @ https://github.com/funilrys/PyFunceble/archive/refs/heads/issue-398.zip
```

----
This poll will stay open until November 3rd, 2024.

**Comment by funilrys** [comment]
Actually, we aren't able to deploy any changes until this situation is resolved either upstream or downstream (here) ...

**Comment by spirillen** [comment]
TBH, I'm against splitting the build, but as psycopg2 are testing for, and requires `pg_conf` (PostgreSQL), I do not see any other option, until this problem have been solved by the third party module.

**Comment by funilrys** [comment]
PostgreSQL dependencies are now optional.
